app/src/main/kotlin/at/bitfire/davdroid/repository/DavCollectionRepository.kt

```diff
@@ -184,6 +187,22 @@ class DavCollectionRepository @Inject constructor(
 
     fun getFlow(id: Long) = dao.getFlow(id)
 
+    /** Returns sync enabled collections for specific authority of given service/account */
+    fun getSyncCollections(serviceId: Long, authority: String) = when (authority) {
+        ContactsContract.AUTHORITY,
+        context.getString(R.string.address_books_authority) ->
+            dao.getByServiceAndSync(serviceId)
+        CalendarContract.AUTHORITY ->
+            dao.getSyncCalendars(serviceId)
+        TaskProvider.ProviderName.JtxBoard.authority ->
+            dao.getSyncJtxCollections(serviceId)
+        TaskProvider.ProviderName.OpenTasks.authority,
+        TaskProvider.ProviderName.TasksOrg.authority ->
+            dao.getSyncTaskLists(serviceId)
+        else ->
+            throw IllegalArgumentException("Invalid authority $authority")
+    }
```

Comment:
I wonder whether dealing with authorities is really the responsibility of the collection repository. Perhaps this would be better done somewhere else.

Reply:
It does make it easier for the abstract Syncer to acquire the correct collections. Otherwise the Syncer implementations will have to get them :)

I will change that now then though